### ISSUE TITLE
[linked editing] Opening HTML tag getting deleted upon typing closing tag

### ISSUE THREAD

**Issue by thdoan** [microsoft/vscode#187117]
Type: <b>Bug</b>

How do I prevent this from happening? I'm manually entering an HTML tag and as soon as I finish typing the closing tag, the opening tag gets deleted:

![Code_vT2yfm3ez4](https://github.com/microsoft/vscode/assets/4110567/cdad3820-19e6-4532-a328-cfde3a9529ed)

VS Code version: Code 1.79.2 (695af097c7bd098fbf017ce3ac85e09bbc5dda06, 2023-06-14T08:57:04.379Z)
OS version: Windows_NT x64 10.0.19045
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz (8 x 2808)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>video_decode: enabled<br>video_encode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled|
|Load (avg)|undefined|
|Memory (System)|31.95GB (16.00GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (14)</summary>

Extension|Author (truncated)|Version
---|---|---
svelte-intellisense|ard|0.7.1
astro-vscode|ast|2.0.18
vscode-tailwindcss|bra|0.9.11
file-properties-viewer|bru|1.0.0
path-intellisense|chr|2.8.4
hex-to-rgba|dak|1.0.0
vscode-eslint|dba|2.4.2
gitlens|eam|14.0.1
vscode-ts-whitespace-converter|joh|1.2.1
l13-diff|L13|1.3.2
vscode-json-validate|rio|1.4.3
svelte-vscode|sve|107.6.0
sort-lines|Tyr|1.10.1
remove-empty-lines|use|1.0.1

</details>

**Comment by aeschli** [comment]
I tried to reproduce:
- new empty profile (no settings, no extensions)
- set settings
```
{
    "html.autoClosingTags": false,
    "editor.linkedEditing": true
}
```
- type `<p>` enter </p>
![Peek 2023-08-18 15-23](https://github.com/microsoft/vscode/assets/6461412/94ea4dcc-51ca-4616-b980-42e1315123ea)


Do you have any other settings that cause the problem?

**Comment by thdoan** [comment]
@aeschli Thanks for looking into this. I can still reproduce the issue on my end, even after disabling all extensions (debug mode) and replacing my `settings.json` with the one you provided above. I think I've identified the root cause: `"editor.linkedEditing": true` -- as soon as I change the value to `false` the issue goes away.

My bad, I should have provided clearer instructions to reproduce the issue:

1. Create a new file with extension `.html` (it needs to be identified as an HTML file in VS Code)
2. Inside this HTML file, add this boilerplate code:

```
<!DOCTYPE html>
<html lang="en">
<body>

</body>
</html>
```

3. On line 4, enter as I did in the recording. Here's the sequence of keys I pressed:

```
<
p
>
Enter
<
/
p
>
```

As soon as I enter the last key (`>`), the opening `<p>` gets changed to `<>` as seen in the recording.
